subsys/bluetooth/host/hci_core.c

```diff
@@ -4890,6 +4912,138 @@ static void le_adv_ext_report(struct net_buf *buf)
 		net_buf_pull(buf, evt->length);
 	}
 }
+
+#if defined(CONFIG_BT_PER_ADV_SYNC)
+
+static struct bt_le_per_adv_sync *get_per_adv_sync(uint16_t handle)
+{
+	for (int i = 0; i < ARRAY_SIZE(per_adv_sync_pool); i++) {
+		if (per_adv_sync_pool[i].handle == handle &&
+		    atomic_test_bit(per_adv_sync_pool[i].flags,
+				     BT_PER_ADV_SYNC_SYNCED)) {
+			return &per_adv_sync_pool[i];
+		}
+	}
+
+	return NULL;
+}
+
+static void le_per_adv_report(struct net_buf *buf)
+{
+	struct bt_hci_evt_le_per_advertising_report *evt;
+	struct bt_le_per_adv_sync *per_adv_sync;
+	struct bt_le_per_adv_sync_recv_info info;
+
+	if (buf->len < sizeof(*evt)) {
+		BT_ERR("Unexpected end of buffer");
+		return;
+	}
+
+	evt = net_buf_pull_mem(buf, sizeof(*evt));
+
+	per_adv_sync = get_per_adv_sync(evt->handle);
+
+	if (!per_adv_sync) {
+		BT_ERR("Unknown handle 0x%04X for per adv report", evt->handle);
+		return;
+	}
+
+	info.tx_power = evt->tx_power;
+	info.rssi = evt->rssi;
+	info.cte = evt->cte;
+	info.addr = &per_adv_sync->params.addr;
+
+	if (per_adv_sync->cb && per_adv_sync->cb->recv) {
+		per_adv_sync->cb->recv(per_adv_sync, &info, &buf->b);
+	}
+}
+
+static void le_per_adv_sync_established(struct net_buf *buf)
+{
+	struct bt_hci_evt_le_per_adv_sync_established *evt =
+		(struct bt_hci_evt_le_per_adv_sync_established *)buf->data;
+	struct bt_le_per_adv_sync_synced_info sync_info;
+
+	if (buf->len != sizeof(*evt)) {
+		BT_ERR("Unexpected event size: Was %u expected %u",
+		       buf->len, (uint32_t)sizeof(*evt));
+		return;
+	}
+
+	if (!syncing_per_adv_sync ||
+	    syncing_per_adv_sync->params.sid != evt->sid ||
+	    bt_addr_le_cmp(&syncing_per_adv_sync->params.addr,
+			   &evt->adv_addr)) {
+		BT_ERR("Unexpected per adv sync established event");
+		return;
+	}
+
+	atomic_set_bit(syncing_per_adv_sync->flags, BT_PER_ADV_SYNC_SYNCED);
+	if (atomic_test_bit(syncing_per_adv_sync->flags,
+			    BT_PER_ADV_SYNC_SCAN_INITIATED)) {
+
+		atomic_clear_bit(syncing_per_adv_sync->flags,
+				 BT_PER_ADV_SYNC_SCAN_INITIATED);
+
+		int err = bt_le_scan_stop();
+
+		if (err) {
+			BT_ERR("Could not stop scan (%d)", err);
+		}
+	}
+
+	syncing_per_adv_sync->handle = evt->handle;
+	syncing_per_adv_sync->interval = evt->interval;
+	syncing_per_adv_sync->clock_accuracy = evt->clock_accuracy;
```

Comment:
evt->clock_accuracy is an enum, but the API has it in uint16_t PPM, either we need convert enum to actual PPM value, or we expose enum to application instead.

Is the clock_accuracy value useful for the application? Maybe we don't need it?

Reply:
Removed it for now. We can add it later if anyone ever needs it.